sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -0,0 +1,730 @@
+ *
+ * @note This API should be used in conjunction with
+ * az_iot_pnp_client_twin_end_property_with_status().
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] json_writer The initialized #az_json_writer to append data to.
+ * @param[in] component_name The name of the component to use with this property payload. If this is
+ * for a root or non-component, this can be #AZ_SPAN_EMPTY.
+ * @param[in] property_name The name of the property to build a response payload for.
+ * @param[in] ack_code The HTTP-like status code to respond with. Please see #az_iot_status for
+ * possible supported values.
+ * @param[in] ack_version The version of the property the application is acknowledging.
+ * @param[in] ack_description An optional description detailing the context or any details about
+ * the acknowledgement. This can be #AZ_SPAN_EMPTY.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p json_writer must not be `NULL`.
+ * @pre \p property_name must be a valid, non-empty #az_span.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The JSON payload was prefixed successfully.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_twin_begin_property_with_status(
+    az_iot_pnp_client const* client,
+    az_json_writer* json_writer,
+    az_span component_name,
+    az_span property_name,
+    int32_t ack_code,
+    int32_t ack_version,
+    az_span ack_description);
+
+/**
+ * @brief End a property response payload with confirmation status.
+ *
+ * @note This API should be used in conjunction with
+ * az_iot_pnp_client_twin_begin_property_with_status().
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] json_writer The initialized #az_json_writer to append data to.
+ * @param[in] component_name The name of the component to use with this property payload. If this is
+ * for a root or non-component, this can be #AZ_SPAN_EMPTY.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p json_writer must not be `NULL`.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The JSON payload was suffixed successfully.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_twin_end_property_with_status(
+    az_iot_pnp_client const* client,
+    az_json_writer* json_writer,
+    az_span component_name);
+
+/**
+ * @brief Read the IoT Plug and Play twin properties version for a specified component.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] json_reader The #az_json_reader to parse through.
+ * @param[in] response_type The #az_iot_pnp_client_twin_response_type representing the message type
+ * associated with the payload.
+ * @param[out] out_version The numeric version of the properties in the JSON payload.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p json_reader must not be `NULL`.
+ * @pre \p out_version must not be `NULL`. It must point to a valid int32_t variable.
```

Comment:
>  It must point to a valid int32_t variable.

What does this mean? Remove.

```suggestion
 * @pre \p out_version must not be `NULL`.
```